src/auth/basic/qgsauthbasicmethod.cpp

```diff
@@ -105,18 +105,28 @@ bool QgsAuthBasicMethod::updateDataSourceUriItems( QStringList &connectionItems,
     return false;
   }
 
+  QString sslMode = QStringLiteral( "prefer" );
```

Comment:
Just a note : according to the PG documentation, "prefer" should be the default anyway if "sslmode" is not set.

Reply:
That's exactly why it's initialized to the default here :)

Reply:
I was just saying that you are not using that value anywhere: a few lines below you check for start with 'verify', so leaving the initial value blank would work just the same.